### PR TITLE
GG-38289 Revert GG-37436 Default value of the statisticsEnabled property of the cache configuration was changed to true.

### PR DESCRIPTION
**Pull request by sk0x50**
https://ggsystems.atlassian.net/browse/GG-38289

This reverts commit 00d71d9b0cff55decb149ea7acf4bc709de096aa.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/configuration/CacheConfiguration.java [hunk 1]
```diff
@@ -531,6 +519,11 @@ public CacheConfiguration(CompleteConfiguration<K, V> cfg) {
         writeBehindFlushSize = cc.getWriteBehindFlushSize();
         writeBehindFlushThreadCnt = cc.getWriteBehindFlushThreadCount();
         writeSync = cc.getWriteSynchronizationMode();
+        storeConcurrentLoadAllThreshold = cc.getStoreConcurrentLoadAllThreshold();
+        maxQryIterCnt = cc.getMaxQueryIteratorsCount();
+        sqlOnheapCache = cc.isSqlOnheapCacheEnabled();
+        sqlOnheapCacheMaxSize = cc.getSqlOnheapCacheMaxSize();
+        evtsDisabled = cc.isEventsDisabled();
```
Comment (sergeyuttsel) [inline modules/core/src/main/java/org/apache/ignite/configuration/CacheConfiguration.java:526]: In the reverted commit, we made this list in alphabetical order. We could save this change.

**Comment by sergeyuttsel** [comment]
When we set the default value of the StatisticsEnabled to true then we discovered the bug that was fixed by https://github.com/gridgain/gridgain/commit/9cc3428ca788bbaf83d6fbe61c421d62bf64e776
This bug is reproduced in CacheOptimisticTransactionsWithFilterTest if the StatisticsEnabled is true. May be we need to set StatisticsEnabled=true explicitly in this test class to test this change.

File: modules/core/src/main/java/org/apache/ignite/configuration/CacheConfiguration.java (hunk 1 above)
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/configuration/CacheConfiguration.java:526]: Yep, makes sense to me. Will do.

**Comment by sk0x50** [comment]
I think creating a new ticket for this case would be the right way to handle it. Will create.

**Review by sergeyuttsel — APPROVED** [review]

**Pull request merged**